R/generate_qc_report.R

```diff
@@ -66,18 +72,40 @@ generate_qc_report <- function(library_id,
     output_dir = dirname(output)
   }
 
+  # create list of parameters
+  params_list <- list(
+    library = library_id,
+    unfiltered_sce = unfiltered_sce,
+    filtered_sce = filtered_sce,
+    processed_sce = processed_sce
+  )
+
+  # define rmd file if none provided
+  if(is.null(rmd_file)){
+    rmd_file <- system.file(file.path("rmd", "qc_report.rmd"), package = "scpcaTools")
+
+    # ignore extra parameters if they exist
+    if(!is.null(extra_params)){
+      warning("`extra_params` will only be used if providing an input rmd file,
+              otherwise these will be ignored.")
```

Comment:
Change to a message, as I don't think it rises to a warning, and moving to a single line to prevent odd formatting of line breaks.
```suggestion
      message("`extra_params` are only used if providing a template file, so will be ignored")
```